### ISSUE TITLE
Feature Request: Allow overriding submitting user via API

### ISSUE THREAD

**Issue by simonbjorzen-ts** [CybercentreCanada/assemblyline#170]
**Is your feature request related to a problem? Please describe.**
We have an use case where we will submit files via API through an external application, but we still want to preserve the submitting user, as that user may log in to AssemblyLine and view the submission there.

**Describe the solution you'd like**
Add a new API permission for impersonating submitters and allow users with that permission to override the submitter field.

**Describe alternatives you've considered**
We are currently using the metadata field to specify the user information, but we'd like to have it in the regular "submitter" field instead.

**Additional context**
The final submitter field appears to be overriden here, no matter what submitter the API submission is sent with: 
https://github.com/CybercentreCanada/assemblyline-ui/blob/93d385b10f6886c571de3e8fba2c21654292347f/assemblyline_ui/api/v4/ingest.py#L328

**Comment by cccs-rs** [comment]
Sounds like a use-case for On Behalf Of (OBO), which we currently have support for within Assemblyline. 
https://github.com/CybercentreCanada/assemblyline-ui/blob/93d385b10f6886c571de3e8fba2c21654292347f/assemblyline_ui/api/v4/authentication.py#L187

This should allow external services to interact with Aseemblyline on behalf of the user using the OBO token.

**Comment by simonbjorzen-ts** [comment]
It looks like that would require using one token per impersonated user if I understand it correctly, which is something we want to avoid.
What we want to do is to have one user with one api token be able to impersonate any user.

Thanks!

**Comment by simonbjorzen-ts** [comment]
Exactly, no need to a full impersonation of the users, just submit/ingest samples as them.

**Comment by cccs-rs** [comment]
This should be properly supported in [4.7.4.10](https://github.com/CybercentreCanada/assemblyline/releases/tag/v4.7.4.stable10).
